### PR TITLE
Share Extension: No more Empty Text

### LINKED ISSUE #5191: Share Extension: Empty text box on share screen
@alisterscott reported that the text box (title/content) on the share screen can appear entirely empty when sharing from an embedded browser in another app. I reproduced that issue when sharing from the embedded browser in Twitter.

### Expected behavior

- Tap a link in another iOS app that opens in an embedded browser (not Safari)
- Share the link to the WordPress app
- The share screen appears with the text box pre-filled with content

### Actual behavior

- Tap a link in another iOS app that opens in an embedded browser (not Safari)
- Share the link to the WordPress app
- The share screen appears with the text box blank

(Sharing the same link from Safari pre-fills the text box as expected.)

##### Tested on WPiOS 6.2.0.20160419

### PR DESCRIPTION
**Pull request by jleandroperez**
Fixes #5191

Testing steps described in the issue itself. Until we implement the JS Extension Add-on, fix consists on:

- Always append the URL, and make it visible
- Linkify the text, whenever the user hits Post

Few things have been prettified here and there.

Needs review: @astralbodies 
Thanks in advance, YO!

### CODE REVIEW

File: WordPress/WordPressShareExtension/ShareViewController.swift [hunk 1]
```diff
@@ -3,32 +3,39 @@ import Social
 import WordPressComKit
 
 
-class ShareViewController: SLComposeServiceViewController {
+class ShareViewController: SLComposeServiceViewController
+{
```
Comment (astralbodies) [inline WordPress/WordPressShareExtension/ShareViewController.swift]: I believe our standard is same line for curly braces on class/struct definitions in Swift.

File: WordPress/WordPressShareExtension/NSExtensionContext+Extensions.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import Foundation
+
+/// Encapsulates NSExtensionContext Helper Methods.
+///
+extension NSExtensionContext
+{
```
Comment (astralbodies) [inline WordPress/WordPressShareExtension/NSExtensionContext+Extensions.swift]: same line :smile:

Comment (jleandroperez) [inline WordPress/WordPressShareExtension/NSExtensionContext+Extensions.swift]: Updated! i could have sworn that for classes / structs, the brace goes always on a new line.
Thanks for the heads up!

File: WordPress/WordPressShareExtension/ShareViewController.swift [hunk 3]
```diff
@@ -59,22 +66,23 @@ class ShareViewController: SLComposeServiceViewController {
     }
     
     override func didSelectPost() {
-        RequestRouter.bearerToken = oauth2Token! as String
+        RequestRouter.bearerToken = oauth2Token!
```
Comment (astralbodies) [inline WordPress/WordPressShareExtension/ShareViewController.swift]: I am beginning to dislike explicit unwrapping even though I used it a lot. Lets add a guard around this and get rid of the `!`.

File: WordPress/WordPressShareExtension/NSExtensionContext+Extensions.swift [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        let urlItemProviders = itemProviders.filter { (itemProvider) in
+            return itemProvider.hasItemConformingToTypeIdentifier(publicUrlIdentifier)
+        }
+        
+        guard urlItemProviders.count > 0 else {
+            completion(nil)
+            return
+        }
+        
+        itemProviders.first!.loadItemForTypeIdentifier(publicUrlIdentifier, options: nil) { (url, error) in
```
Comment (astralbodies) [inline WordPress/WordPressShareExtension/NSExtensionContext+Extensions.swift]: Switch this to `urlItemProviders.first!` or iterate through until one of the providers can load the URL. I tried sharing an item from News.app and the first item provider was for the plain text of the article.

File: WordPress/WordPressShareExtension/ShareViewController.swift [hunk 5]
```diff
@@ -139,6 +147,17 @@ class ShareViewController: SLComposeServiceViewController {
         pushConfigurationViewController(pickerViewController)
     }
     
+    private func loadTextViewContent() {
+        extensionContext?.loadWebsiteUrl { url in
+            dispatch_async(dispatch_get_main_queue()) {
+                let current = self.contentText ?? String()
+                let source  = url?.absoluteString ?? String()
+                
+                self.textView.text = "\(current)\n\n\(source)"
```
Comment (astralbodies) [inline WordPress/WordPressShareExtension/ShareViewController.swift]: If no URL or only a URL exists lets make sure no blank space is put into the text box.
img_9856

File: WordPress/WordPressShareExtension/ShareViewController.swift (hunk 3 above)
Comment (jleandroperez) [inline WordPress/WordPressShareExtension/ShareViewController.swift]: Done, thanks!

File: WordPress/WordPressShareExtension/NSExtensionContext+Extensions.swift (hunk 4 above)
Comment (jleandroperez) [inline WordPress/WordPressShareExtension/NSExtensionContext+Extensions.swift]: OMG this is clearly an error!! `urlItemProviders` isn't being referenced after filtering!. Nice catch!

**Comment by astralbodies** [comment]
Changes look good, @jleandroperez. :shipit:

**Comment by jleandroperez** [comment]
Thank you @astralbodies!

**Pull request merged**